### PR TITLE
fix: skip reconcile when UserSignup is being deleted

### PR DESCRIPTION
**Pull request by MatousJobanek**
When UserSignup is being deleted, then (in my opinion) we should not do anything in the reconcile. 
My guess is that the flakiness is caused by reconciliation that creates MUR just after their deletion - this is at least what it looks like from the logs. 
I can be wrong, but still - when the UserSignup is being deleted, we should just exit the reconcile.
I'm open to discussion ;-)

### CODE REVIEW

File: controllers/usersignup/usersignup_controller.go [hunk 1]
```diff
@@ -137,6 +138,11 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	}
 	logger = logger.WithValues("username", userSignup.Spec.Username)
 
+	if util.IsBeingDeleted(userSignup) {
+		logger.Error(err, "The UserSignup is being deleted")
```
Comment (alexeykazakov) [inline controllers/usersignup/usersignup_controller.go:142]:
```suggestion
		logger.Info("The UserSignup is being deleted")
```

**Review by alexeykazakov — APPROVED** [review]
Yeah.. this might be the reason... especially when we delete the UserSingup with Foreground deletion option which holds the UserSignup alive (with the deletion timestamp) until the MUR is actually deleted.
Good catch.

File: controllers/usersignup/usersignup_controller.go (hunk 1 above)
Comment (MatousJobanek) [reply controllers/usersignup/usersignup_controller.go]: copy-paste - thanks :-)

**Pull request merged**